### PR TITLE
Release v0.3.0

### LINKED ISSUE #18: get_wallet_transactions: transaction_types filter silently drops some valid values (deposit, liquidation_fee) — returns 0 despite matching rows existing
## Summary

The `transaction_types` filter on `get_wallet_transactions` returns **zero** results for certain valid values even when rows of that exact type exist in the unfiltered response. Confirmed affected: `deposit` and `liquidation_fee`. Other values (`funding`, `commission`, `cashflow`, `sub_account_transfer`) filter correctly. So specific type values are silently dropped/unmatched rather than filtered.

For LLM-driven clients this is a correctness trap: "show my deposits" maps to `transaction_types=["deposit"]`, returns nothing, and the assistant reports "no deposits" on an account that has many.

## Reproduction

Single account, same key, back-to-back calls. `total_count` reported by `meta`:

| `transaction_types` | total_count | type present in unfiltered data? | result |
|---|---|---|---|
| `["funding"]` | 124 | yes | ✅ correct |
| `["commission"]` | 114 | yes | ✅ correct |
| `["cashflow"]` | 75 | yes | ✅ correct |
| `["sub_account_transfer"]` | 1 | yes | ✅ correct |
| `["deposit"]` | 0 | **yes (7 deposits)** | ❌ wrong |
| `["liquidation_fee"]` | 0 | **yes (≥2)** | ❌ wrong |
| `["withdrawal"]` | 0 | none on this account | inconclusive |

**Proof the dropped types exist** — fetched UNFILTERED over a date window and counted `transaction_type` values:
```
get_wallet_transactions(start_time_us=<...>, end_time_us=<...>, page_size=200)
→ type counts include: { ..., deposit: 4, liquidation_fee: ... }
```
Example deposit row actually present (redacted):
```json
{ "transaction_type": "deposit", "amount": "117.64705882", "asset_symbol": "USD",
  "meta_data": { "deposit_id": <int>, "transaction_id": "<bank ref>" },
  "created_at": "2025-11-21T07:10:03Z" }
```
The stored value is literally `"deposit"`, yet `["deposit"]` matches nothing.

## Expected behavior

`transaction_types=[X]` returns all rows where `transaction_type == X`, for every value that can appear on a row — consistent with the values that already work.

## Impact

- False-empty results for affected types → "no deposits" / "no liquidation fees" when they exist. Dangerous for accounting, tax, reconciliation.
- LLM clients take the empty filtered result at face value and report it as fact.

## Suspected cause

The filter appears to use an allowlist / enum mapping that omits `deposit` and `liquidation_fee` (and possibly others). Recommend auditing the full set of accepted filter values against the complete set of `transaction_type` values the ledger can emit, and adding a test that, for each distinct type seen unfiltered, the single-type filter returns a matching non-zero count.

## Workaround

Fetch unfiltered (optionally windowed via `start_time_us`/`end_time_us`) and filter on `transaction_type` client-side.

## Environment

- API: `https://api.india.delta.exchange/v2`
- Endpoint: `GET /v2/wallet/transactions`
- MCP tool: `get_wallet_transactions`

### PR DESCRIPTION
**Pull request by anuj-delta**
Release v0.3.0 — brings develop to main.

## Features

- **Opt-in debug logging** (#19): `DELTA_MCP_DEBUG=1` writes HTTP request URLs (incl. filter params) and response bodies to `~/.delta-exchange-mcp/logs/debug-<ts>-<pid>.log`. New `get_debug_status` tool reports the log path; path also printed in the startup banner. API key/secret/signature are never logged. `DELTA_MCP_DEBUG_FILE` overrides the path.

## Fixes

- **Surface the API's default ~90-day window** (#20, closes #18): the backend silently defaults `start_time` to ~90 days when omitted, so historical/sparse queries (e.g. deposits) under-reported and looked like a broken `transaction_types` filter. `get_wallet_transactions`, `get_fills`, and `bulk_fills_export` now attach a `notice` when `start_time_us` is omitted, and their docstrings state that an explicit `start_time_us` is required for older/full history.

## Version

`pyproject.toml` + `uv.lock` bumped 0.2.0 → 0.3.0 (minor: new env vars/tool + behavior change).

After merge: tag `v0.3.0`, publish to PyPI, draft the GitHub release per RELEASING.md.

### CODE REVIEW

File: src/delta_exchange_mcp/debug_log.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+def _open_handler(path: Path) -> tuple[logging.FileHandler, Path]:
+    """Create the dir and a FileHandler, falling back to a tempdir on OSError."""
+    try:
+        path.parent.mkdir(parents=True, exist_ok=True)
+        return logging.FileHandler(path, encoding="utf-8"), path
+    except OSError:
+        fallback = Path(tempfile.gettempdir()) / "delta-exchange-mcp" / path.name
+        fallback.parent.mkdir(parents=True, exist_ok=True)
+        return logging.FileHandler(fallback, encoding="utf-8"), fallback
```
[the remark this answers is not in the document]
Comment (anuj-delta) [reply src/delta_exchange_mcp/debug_log.py]: Fixed in ba84ba4 (merged to develop via #22, now on this release branch). The log file is chmod 0o600 immediately after creation (file is empty at that point, before any body is written); best-effort, no-op on Windows. Regression test asserts mode 0600 on POSIX.

**Comment by anuj-delta** [comment]
On the Greptile summary note that `get_order_history` is not annotated like the other three: this is intentional, not an oversight. Live API probing showed `/orders/history` does **not** share the ~90-day default behaviour:

- No time range → returns full history (in testing, 111 rows back to January, not truncated to 90 days).
- A wide explicit `start_time` → returns **fewer** rows (8, a single day), not more.

So adding a "limited to ~90 days, pass start_time_us for more" notice there would be factually wrong, and passing `start_time` actually returns less. The `/orders/history` time-filter behaviour looks like a separate upstream bug and is being raised with the platform team rather than papered over with a misleading notice.
